ttnn/cpp/ttnn/tensor/types.hpp

```diff
@@ -706,6 +785,23 @@ struct Shape {
         return Shape{tt::tt_metal::LegacyShape{this->value, tt::tt_metal::Padding{this->value.rank()}}};
     }
 
+    SimpleShape padded_shape() const {
+        std::vector<uint32_t> values(this->rank());
```

Comment:
you didn't reserve, but allocated this->rank() elements
use `values[i] = this.value[i];` or change to reserve values

Reply:
fixed, thank you